extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoP.java

```diff
@@ -258,20 +254,13 @@ void connect(MongoClient newClient, boolean snapshotsEnabled) {
             try {
                 logger.fine("(Re)connecting to MongoDB");
                 if (databaseName != null) {
-                    if (throwOnNonExisting) {
```

Comment:
did we loose the ability to check existence after reconnection? Or is the log `(Re)connecting to MongoDB` misleading? `ALWAYS` checks only at the start of the query.

Reply:
Yes, that's the tradeoff unfortunatelly

Reply:
ok, I refactored it completely restoring also old behaviour ;)